addons/stock/models/stock_picking.py

```diff
@@ -921,6 +921,11 @@ def button_validate(self):
         pickings_to_backorder.with_context(cancel_backorder=False)._action_done()
         return True
 
+    def action_set_quantities_to_reservation(self):
+        for move in self.move_lines.filtered(lambda move: move.state in ('partially_available', 'assigned')):
+            if move.product_id.tracking == 'none':
+                move.quantity_done = move.reserved_availability
```

Comment:
In case the stock move has 2 (or more) stock move lines, you will get this error : 
https://github.com/odoo/odoo/blob/7844ed0a5fcf4a2139c355b8e640ba50ec818527/addons/stock/models/stock_move.py#L323-L331

I think the best would be to loop on `move.line_ids` and set the qty_done line by line